jerry-core/jerry.c

```diff
@@ -2087,3 +2087,46 @@ jerry_dispatch_object_free_callback (ecma_external_pointer_t freecb_p, /**< poin
 
   jerry_make_api_available ();
 } /* jerry_dispatch_object_free_callback */
+
+/**
+ * Reserve a slot in the context for a module
+ *
+ * Returns: 0 or positive on success, -1 on failure.
+ */
+
+/* This value is context-independent */
+static int module_index = -1;
```

Comment:
Yeah, in the proposal that arose from #1717, module indices are global (and assigned once per boot-up).
That means that if multiple contexts use different sets of modules, there will be unused slots in the `module_data` array. I think this is probably acceptable.

The alternative is to add another per-context module registration step and another level of indirection. That would also add extra memory, so I'm not convinced that at the bottom line this would be better compared to having a couple unused slots.